src/observable/DeferObservable.ts

```diff
@@ -14,11 +12,20 @@ export class DeferObservable<T> extends Observable<T> {
   }
 
   protected _subscribe(subscriber: Subscriber<T>) {
-    const result = tryCatch(this.observableFactory)();
-    if (result === errorObject) {
-      subscriber.error(errorObject.e);
-    } else {
+    const result = callObservableFactory(this.observableFactory, subscriber);
+    if (result !== null) {
       result.subscribe(subscriber);
     }
   }
 }
+
+function callObservableFactory<T>(factory: () => Observable<T>, errorSubscriber: Subscriber<T>): Observable<T> {
```

Comment:
Can you try adding this as a member of DeferObservable? That way you don't create a closure in `_subscribe` to get at `callObservableFactory`. It probably won't make a huge difference, but I wanted to bring it up.